datumaro/plugins/data_formats/segment_anything/exporter.py

```diff
@@ -0,0 +1,154 @@
+# Copyright (C) 2023 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+
+
+import logging as log
+import os
+import os.path as osp
+from collections import defaultdict
+
+from pycocotools import mask as mask_utils
+
+from datumaro.components.annotation import AnnotationType
+from datumaro.components.errors import DatumaroError, MediaTypeError
+from datumaro.components.exporter import Exporter
+from datumaro.components.media import Image
+from datumaro.util import NOTSET, dump_json_file
+
+
+def replace(json_data, key, value_new):
+    value_origin = json_data[key]
+    if value_origin is NOTSET:
+        json_data[key] = value_new
+        return
+    if value_origin != value_new:
+        raise DatumaroError(f"The value for '{key}' is not same for item {value_new}")
+
+
+class SegmentAnythingExporter(Exporter):
+    DEFAULT_IMAGE_EXT = ".jpg"
+
+    def __init__(
+        self,
+        extractor,
+        save_dir,
+        **kwargs,
+    ):
+        super().__init__(extractor, save_dir, **kwargs)
+
+    def apply(self):
+        if self._extractor.media_type() and not issubclass(self._extractor.media_type(), Image):
+            raise MediaTypeError("Media type is not an image")
+
+        os.makedirs(self._save_dir, exist_ok=True)
+
+        subsets = self._extractor.subsets()
+        pbars = self._ctx.progress_reporter.split(len(subsets))
+
+        _image_ids = set()
+
+        image_id = 1
+        for pbar, (subset_name, subset) in zip(pbars, subsets.items()):
+            for item in pbar.iter(subset, desc=f"Exporting {subset_name}"):
+                try:
+                    # make sure file_name is flat
+                    file_name = self._make_image_filename(item).replace("/", "__")
+                    image_id = int(item.attributes.get("id", image_id))
+                    while image_id in _image_ids:
+                        image_id += 1
+                    _image_ids.add(image_id)
+
+                    height, width = item.media.size
+                    json_data = {
+                        "image": {
+                            "image_id": image_id,
+                            "file_name": file_name,
+                            "height": height,
+                            "width": width,
+                        },
+                        "annotations": [],
+                    }
+
+                    annotations_grouped = defaultdict(lambda: [])
+                    for annotation in item.annotations:
+                        annotations_grouped[annotation.id].append(annotation)
```

Comment:
Why do you group by `id`, not `group`? To my knowledge, `group` is a field to represent the instance label. Please refer to
https://github.com/openvinotoolkit/datumaro/blob/1c150936f2eb14189b1f48263580ac3b24f58f61/datumaro/util/annotation_util.py#L15-L24

Reply:
Annotations are allowed to have duplicated `id` and a single 'id' represents a single object in the image. The annotations with the same `id` indicates that they represent the same single object in the image just like COCO dataset where an object could be represented by `Point`, `Polygon`, `Mask`, `Bbox`, etc.
In Segment Anything data format, annotation types are `Bbox` **and** `Mask`.
Here, we are grouping annotations by the same id to gather all annotations representing the same object.

Reply:
1. My point is that such a characteristic of Segment Anything data format (same `id` for same instance) should be reflected as `group` in Datumaro side because it has been Datumaro's design philosophy. It is very important to quality that the entire code adheres to the common design philosophy. I guess you might already consider this here.
https://github.com/openvinotoolkit/datumaro/blob/da29cc109f54ac6df2c28424a431b56e2355c0da/datumaro/plugins/data_formats/segment_anything/base.py#L124

2. You cannot assume that the input dataset of `SegmentAnythingExporter` will always come from `SegmentAnythingBase`. Under the constraint that an instance should be represented by `group`, to handle arbitrary dataset inputs as possible, we think that a merge process is necessary, such as the COCO instance exporter.

Reply:
I get what you mean. Fixed in https://github.com/openvinotoolkit/datumaro/pull/1005/commits/235e5a9bb05595dfbe18cdb00211a700323dd63b

Reply:
Annotations are merged in https://github.com/openvinotoolkit/datumaro/pull/1005/commits/dc0ac4f50f1f0fc9a989c59d688225ee831b53cd